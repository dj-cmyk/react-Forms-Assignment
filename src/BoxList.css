
.BoxList-form {
    margin: auto;
    width: 75%;
}

.BoxList-box {
    display: flex;
    flex-wrap: wrap;
}