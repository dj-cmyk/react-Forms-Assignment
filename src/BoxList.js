import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import "./BoxList.css"


const BoxList = () => {
    const INITIAL_STATE = []
      const [boxes, setBoxes] = useState(INITIAL_STATE);
      
      const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, { ...newBox, id: uuid() }])
      }

      const removeBox = (id) => {
          setBoxes(boxes => boxes.filter(box => box.id !== id))
      }

      return (
        <div className="BoxList">
          <h3 className="BoxList-header">BoxList</h3>
          <div className="BoxList-form">
            <NewBoxForm addBox={addBox} />
          </div>
          <div className="BoxList-box">
            {boxes.map(({ id, color, height, width }) => <Box color={color} height={height} width={width} key={id} id={id}removeBox={removeBox}/>)}
          </div>
    
        </div>
      )
}



export default BoxList;
