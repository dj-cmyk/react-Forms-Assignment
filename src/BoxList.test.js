import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

//smoke test 
it("renders without crashing", function() {
  render(<BoxList />);
});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});


//creates new box test
it("can add a new box", function() {
  const { getByLabelText, queryByText } = render(<BoxList />);

  // no box in document to start (remove button has text "X")
  expect(queryByText("X")).not.toBeInTheDocument();

  const colorInput = getByLabelText("Color:");
  const heightInput = getByLabelText("Height:");
  const widthInput = getByLabelText("Width:");
  const submitBtn = queryByText("Add Box")

  // fill out the form
  fireEvent.change(colorInput, { target: { value: "#555555" }});
  fireEvent.change(heightInput, { target: { value: 100 }});
  fireEvent.change(widthInput, { target: { value: 100 }});
  fireEvent.click(submitBtn);

  // box in document
  expect(queryByText("X")).toBeInTheDocument();
});


//removes box test
it("can remove box", function() {
  const { getByLabelText, queryByText } = render(<BoxList />);

  const colorInput = getByLabelText("Color:");
  const heightInput = getByLabelText("Height:");
  const widthInput = getByLabelText("Width:");
  const submitBtn = queryByText("Add Box");

  // fill out the form
  fireEvent.change(colorInput, { target: { value: "#555555" }});
  fireEvent.change(heightInput, { target: { value: 100 }});
  fireEvent.change(widthInput, { target: { value: 100 }});
  fireEvent.click(submitBtn);

  // box in document
  expect(queryByText("X")).toBeInTheDocument();

  // click on X button to remove the box
  const removeBtn = queryByText("X");
  fireEvent.click(removeBtn)

  expect(queryByText("X")).not.toBeInTheDocument();
});