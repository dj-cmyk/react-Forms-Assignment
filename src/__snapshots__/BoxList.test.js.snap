// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h3
      class="BoxList-header"
    >
      BoxList
    </h3>
    <div
      class="BoxList-form"
    >
      <form
        class="NewBoxForm"
      >
        <label
          for="color"
        >
          Color: 
        </label>
        <input
          class="NewBoxForm-input"
          id="color"
          name="color"
          type="color"
          value="#000000"
        />
        <label
          for="height"
        >
          Height: 
        </label>
        <input
          class="NewBoxForm-input"
          id="height"
          max="1000"
          min="10"
          name="height"
          type="number"
          value=""
        />
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          class="NewBoxForm-input"
          id="width"
          max="1000"
          min="10"
          name="width"
          type="number"
          value=""
        />
        <button>
          Add Box
        </button>
      </form>
    </div>
    <div
      class="BoxList-box"
    />
  </div>
</DocumentFragment>
`;
